pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -264,31 +267,40 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) (roachpb.Key
 		return nil, err
 	}
 
+	// extend determines if the new span should be added to the last span. This
+	// must occur if the previous span ended mid row, or if the new span is a
+	// continuation of the previous span (i.e. the new span picks up where the
+	// previous one ended and has the same time bounds).
+	var extend bool
+	last := s.flushedFiles[len(s.flushedFiles)-1]
+	if s.midRow {
+		extend = true
+	} else if len(s.flushedFiles) > 0 {
+		extend = last.Span.EndKey.Equal(span.Key) &&
+			last.EndTime.EqOrdering(resp.metadata.EndTime) &&
+			last.StartTime.EqOrdering(resp.metadata.StartTime) &&
+			last.EntryCounts.DataSize < fileSpanByteLimit
+	}
+
 	if len(resp.resumeKey) > 0 {
 		span.EndKey, s.midRow = adjustFileEndKey(span.EndKey, maxKey, maxRange)
 		// Update the resume key to be the adjusted end key so that start key of the
 		// next file is also clean.
 		resp.resumeKey = span.EndKey
-	} else {
-		s.midRow = false
-	}
-
-	// If this span extended the last span added -- that is, picked up where it
-	// ended and has the same time-bounds -- then we can simply extend that span
-	// and add to its entry counts. Otherwise we need to record it separately.
-	if l := len(s.flushedFiles) - 1; l >= 0 && s.flushedFiles[l].Span.EndKey.Equal(span.Key) &&
-		s.flushedFiles[l].EndTime.EqOrdering(resp.metadata.EndTime) &&
-		s.flushedFiles[l].StartTime.EqOrdering(resp.metadata.StartTime) {
-		s.flushedFiles[l].Span.EndKey = span.EndKey
-		s.flushedFiles[l].EntryCounts.Add(resp.metadata.EntryCounts)
-		s.flushedFiles[l].ApproximatePhysicalSize += resp.metadata.ApproximatePhysicalSize
+	}
+
+	if extend {
+		last.Span.EndKey = span.EndKey
+		last.EntryCounts.Add(resp.metadata.EntryCounts)
+		last.ApproximatePhysicalSize += resp.metadata.ApproximatePhysicalSize
```

Comment:
`last` here is a copy by value so this isn't incrementing the one in `flushedFiles` is it?